### PR TITLE
[Core][GidIO] Clean up mNodeList related operations

### LINKED ISSUE #2595: [Core][GiDIO] Not useful loop
https://github.com/KratosMultiphysics/Kratos/blob/23294d189df6a593cf107e7eb2c705c70b30b93e/kratos/includes/gid_io.h#L1406

@KratosMultiphysics/technical-committee This loop is doing nothing except expending time :(

### PR DESCRIPTION
**Pull request by loumalouomega**
Unused

Fixes #2595

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping

File: kratos/includes/gid_io.h [hunk 1]
```diff
@@ -1235,7 +1230,6 @@ class GidIO : public IO
         {
             nodes_id[0] = node_iterator->Id();
             GiD_fWriteSphereMat(mMeshFile, node_iterator->Id(), nodes_id[0], node_iterator->FastGetSolutionStepValue(radius), node_iterator->FastGetSolutionStepValue(particle_material));
-//             mNodeList.push_back(*node_iterator);
```
Comment (roigcarlo) [inline kratos/includes/gid_io.h:1238]: We could probably remove all the commented loop?
Seems related to DEM, @maceligueta ?

File: kratos/includes/gid_io.h [hunk 2]
```diff
@@ -1335,7 +1329,6 @@ void WriteClusterMesh( MeshType& rThisMesh )
         {
             nodes_id[0] = node_iterator->Id();
             GiD_fWriteClusterMat(mMeshFile, node_iterator->Id(), nodes_id[0], node_iterator->FastGetSolutionStepValue(radius), node_iterator->FastGetSolutionStepValue(particle_material));
-//             mNodeList.push_back(*node_iterator);
```
Comment (roigcarlo) [inline kratos/includes/gid_io.h:1338]: Same here

**Review by roigcarlo — COMMENTED** [review]
Rest of the changes ok

**Comment by loumalouomega** [comment]
Done

**Review by roigcarlo — APPROVED** [review]
Good to me

**Pull request merged**